### PR TITLE
End cleanup on rustdoc-js tools

### LINKED ISSUE #70689: Merge JS testers into one
The ground work was started in #70224. It needs to go further and just merge both scripts into one. Which means that we'll have to rework a bit the std checks in order to provide a folder.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #70689.

It removes a lot of code, which is pretty nice. :3

We now only have one entry point and a small set of options.

r? @ollie27

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
The failure seems to come from outside of my code. An idea maybe @pietroalbini ?

**Comment by pietroalbini** [comment]
If I remember correctly that was just a spurious failure. Restarted the PR build.

**Comment by GuillaumeGomez** [comment]
And it passed! \o/

cc @ollie27 (also @kinnison ;) )

**Comment by GuillaumeGomez** [comment]
Updated!

File: src/tools/compiletest/src/runtest.rs [hunk 1]
```diff
@@ -2807,10 +2807,16 @@ impl<'test> TestCx<'test> {
...
 
             let root = self.config.find_rust_src_root().unwrap();
+            let file_stem =
+                self.testpaths.file.file_stem().and_then(|f| f.to_str()).expect("no file stem");
             let res = self.cmd2procres(
                 Command::new(&nodejs)
                     .arg(root.join("src/tools/rustdoc-js/tester.js"))
-                    .arg(out_dir.parent().expect("no parent"))
+                    .arg("--doc-folder")
+                    .arg(out_dir.parent().expect("no parent").join(file_stem.replace(".js", "")))
```
Comment (ollie27) [inline src/tools/compiletest/src/runtest.rs:2816]:
```suggestion
                    .arg(out_dir)
```

File: src/tools/compiletest/src/runtest.rs [hunk 2]
```diff
@@ -2807,10 +2807,16 @@ impl<'test> TestCx<'test> {
...
+                    .arg("--crate-name")
+                    .arg(file_stem)
```
Comment (ollie27) [inline src/tools/compiletest/src/runtest.rs:2818]:
```suggestion
                    .arg(file_stem.replace("-", "_"))
```

**Review by ollie27 — COMMENTED** [review]
This is looking good now, thanks. With these minor issues fixed, r=me.

**Pull request merged**